ruby/ext/google/protobuf_c/defs.c

```diff
@@ -173,16 +173,32 @@ VALUE DescriptorPool_add(VALUE _self, VALUE def) {
 
 /*
  * call-seq:
- *     DescriptorPool.build(&block)
+ *     DescriptorPool.build(file_descriptor = nil, &block)
  *
  * Invokes the block with a Builder instance as self. All message and enum types
  * added within the block are committed to the pool atomically, and may refer
  * (co)recursively to each other. The user should call Builder#add_message and
  * Builder#add_enum within the block as appropriate.  This is the recommended,
  * idiomatic way to define new message and enum types.
+ * If you do not pass a file_descriptor, a default one will be set with proto3.
+ * This is to preserve backwards compatibility with prior code that only supported proto3.
  */
-VALUE DescriptorPool_build(VALUE _self) {
-  VALUE ctx = rb_class_new_instance(0, NULL, cBuilder);
+VALUE DescriptorPool_build(int argc, VALUE* argv, VALUE _self) {
+  VALUE file_descriptor_rb;
+  if (argc > 1) {
+    rb_raise(rb_eArgError, "Expected 0 or 1 argument, got %d", argc);
+  } else if (argc == 1) {
+    if (CLASS_OF(argv[0]) != cFileDescriptor) {
```

Comment:
Something another coworker pointed out, should this be `CLASS_OF(argv[0]) != cFileDescriptor` or should it be calling the Ruby `rb_obj_is_kind_of` or `rb_obj_is_instance_of` instead? I was mostly mirroring the existing code you had.